src/atopile/front_end.py

```diff
@@ -490,16 +525,33 @@ def _build(self, context: Context, ref: Ref) -> L.Node:
             raise errors.UserKeyError.from_ctx(
                 context.scope_ctx, f'No declaration of "{ref}" in {context.file_path}'
             )
-        return self._init_node(context.scope_ctx, ref)
+        try:
+            return self._init_node(context.scope_ctx, ref)
+        except* SkipPriorFailedException:
+            raise errors.UserException("Build failed")
+        finally:
+            self._finish()
 
     def _finish(self):
-        with ExceptionAccumulator() as ex_acc:
+        with ExceptionAccumulator(
+            errors._BaseBaseUserException, SkipPriorFailedException
+        ) as ex_acc:
             for param, (value, ctx) in self._param_assignments.items():
                 with ex_acc.collect(), ato_error_converter(), log_user_errors(logger):
                     if value is None:
-                        raise errors.UserKeyError.from_ctx(
+                        ex = errors.UserKeyError.from_ctx(
                             ctx, f"Parameter {param} never assigned"
                         )
+                        if param in self._promised_params:
```

Comment:
Again, std lib issue. Don't complain unless we actually try use an undefined param